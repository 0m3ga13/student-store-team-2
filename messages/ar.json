{
    "Index": {
        "Title": "المتجر الصغير",
        "Home": "الرئيسية",
        "Products": "المنتجات",
        "Orders": "الطلبيات",
        "State": "الحالة",
        "Donate": "تبرع",
        "Buy": "شراء",
        "Borrow": "إعارة",
        "Description": "هذه المنصة هي مربحة للجميع: القرضاء يحققون دخلاً إضافيًا، والمقترضون يقللون التكاليف.",
        "Profile": "الحسـاب",
        "For Sale": "للبيع",
        "For Borrow": "للإعارة",
        "Sale & Borrow": "إعارة وشراء",
        "Day": "لليوم",
        "How It Works": "طريـقة العمل",
        "recently added": "المنتوجات المضافة حديثا",
        "Hello": "مرحبا",
        "Sign out": "تسجيل الخروج",
        "Sign In": "تسجيل الدخول",
        "Login with Google": "سحل الدخول بحسابك",
        "Sign Up With Google": "سحل الدخول بحسابك",

        "Free delivery":"توصيل مجاني",
        "feature descriprion 1":"يمكنك استلام طلبك دون تحمل أي تكاليف شحن إضافية",
        "qualite guarantee":"جودة مضمونة",
        "feature descriprion 2":"منتجاتنا أو خدماتنا ذات معايير عالية وتعمل كما وعدنا بها",
        "daily offers":"عروض يومية",
        "feature descriprion 3":"خصومات وعروض ترويجية حصرية يوميا",
        "100% secure payement":"دفع آمن %100",
        "feature descriprion 4":"معلوماتك المالية محمية ومعاملاتك آمنة من الوصول غير المصرح به",

        "My Dashboard": "لوحة المعلومات",
        "Awaiting Pickup": "في انتظار الاستلام",
        "All": "الكل",
        "Active": "فعال",
        "Completed": "مكتمل",
        "Sort": ":صنف",
        "Newest": "الأحدث",
        "Oldest": "الأقدم",
        "borrowings": "القروض",
        "lendings": "الإقراضات",
        "purchases": "المشتريات",
        "sales": "المبيعات",
        "My Borrowings": "قروضي",
        "My Lendings": "اقراضاتي",
        "My Purchases": "مشترياتي",
        "My Sales": "مبيعاتي",
        "General details": "المعلومات العامة",
        "First Name": "الإسم",
        "Last Name": "اللقب",
        "Birth Date": "تاريخ الميـلاد",
        "Gender": "الجنس",
        "Institution": "معهد الدراسة",
        "Contact & address Detail": "معلومات الاتصال ",
        "Phone Number": "رقم الهـاتف",
        "Building": "رقم العمارة",
        "Street": "شارع",
        "City": "مدينة",
        "State_place": "ولاية",
        "Save Changes": "حفظ التغييرات",
        "Upload Image": "تحميل الصورة",
        "Close": "غلق",
        "Upload": "تحميل",
        "Loading": "يجري  التحميل",
        "Not a member ?": "لست عضو؟",
        "Sign Up": "تسجيل",
        "Store": "متجرك",
        "Create Your": "إصنع",
        "and start exchanging with other students": "وابدأ التبادل مع الطلاب الآخرين",
        "Or:": ":أو",
        "Login": "دخـول",
        "LOGIN": "دخـول ",
        "Already a member ?": "هل أنت عضو؟",
        "Email and password are required": "البريد الإلكتروني وكلمة المرور مطلوبان.",
        "Password should be at least 8 characters long": "يجب أن تكون كلمة المرور على الأقل 8 أحرف.",
        "and Conditions and Privacy Policy of The website": "وشروط سياسة الخصوصية للموقع",
        "Email": "البريد الإلكتروني",
        "Password": "كلمة المرور",
        "Full Name": "الاسم الكامل",
        "Address": "العنوان",
        "and Privacy Policy": "وسياسة الخصوصية",
        "I have read and agree with all of the Terms": "لقد قرأت وأوافق على جميع الشروط",
        "SECURITY": "الحماية",
        "PROFILE": "الحساب",
        "Upload ID": "تحميل الهوية ",
        "Edit ID": "تعديل الهوية ",
        "Edit Image": "تعديل الصورة",
        "WHY DONATE": "لماذا أتبرع",
        "MAKE A DIFFERENCE BY DONATING": "إصنع الفارق من خلال التبرع",
        "Donate paragraph": ".في حياتنا سريعة الوتيرة، من السهل أن ننسى قوة الإمتنان. يعتمد الطلاب الذين غالبًا ما تكون لديهم ميزانيات محدودة على دعمنا. تبرعك هو أكثر من مجرد مساعدات مالية؛ إنه رمز للامتنان. من خلال العطاء بسخاء، أنت تساعد الطلاب على خلق مستقبل أكثر إشراقًا، بمساهمة واحدة في كل مرة",
        "Small Help": "مساعدة صغيرة",
        "Some Help": "بعض المساعدة",
        "Big Help": "مساعدة كبيرة",
        "List an Item": "إضافة منتوج",
        "Drag and drop some files here or click to select files": "قم بسحب ووضع بعض الملفات هنا، أو انقر لتحديد الملفات",
        "Product Name": "إسم المنتوج",
        "Location": "الموقع",
        "Category": "الصنف",
        "books": "الكتب",
        "kitchen": "المطبخ",
        "sports": "الـرياضة",
        "games": "الألعاب",
        "clothes": "مـلابس",
        "electronics": "إلكترونيات",
        "furniture": "أثاث",
        "description": "وصف",
        "Product Condition": "حالة المنتج",
        "Sell": "بيع",
        "Price Per Week": "السعر للأسبوع",
        "Price": "السعر",
        "Upload Image(s)": "إصافة الصور",
        "List": "إضافة"

    }
}
