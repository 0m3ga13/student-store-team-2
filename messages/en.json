{
    "Index": {
        "Title": "MiniStore",
        "Home": "Home",
        "Products": "Products",
        "Orders": "Orders",
        "State": "State",
        "Donate": "Donate",
        "Buy": "Buy",
        "Borrow": "Borrow",
        "Description": "This platform is a win-win for everyone involved lenders earn extra income, borrowers cut costs.",
        "Profile": "Profile",
        "for_sale": "for_sale",
        "for_borrow": "for_borrow",
        "Sale & Borrow": "Sale & Borrow",
        "Day": "Day",
        "How It Works": "How It Works",
        "recently added": "recently added",
        "Hello": "Hello",
        "Sign out": "Sign out",
        "Sign In": "Sign In",
        "Login with Google": "Login with Google",
        "Sign Up With Google": "Sign Up With Google",
        "Free delivery": "Free delivery",
        "feature descriprion 1": "You can receive your order without incurring any additional shipping costs",
        "qualite guarantee": "qualite guarantee",
        "feature descriprion 2": "Our products or services meet high standards and perform as promised.",
        "daily offers": "daily offers",
        "feature descriprion 3": "Exclusive discounts and promotions on a daily basis",
        "100% secure payement": "100% secure payement",
        "feature descriprion 4": "Your financial information is protected and your transactions are safe from unauthorized access",
        "My Dashboard": "My Dashboard",
        "Awaiting Pickup": "Awaiting Pickup",
        "All": "All",
        "Active": "Active",
        "Completed": "Completed",
        "Sort": "Sort:",
        "Newest": "Newest",
        "Oldest": "Oldest",
        "my stuff":"my stuff",
        "borrowings": "borrowings",
        "lendings": "lendings",
        "purchases": "purchases",
        "sales": "sales",
        "My Borrowings": "My Borrowings",
        "My Lendings": "My Lendings",
        "My Purchases": "My Purchases",
        "My Sales": "My Sales",
        "General details": "General details",
        "First Name": "First Name",
        "Last Name": "Last Name",
        "Birth Date": "Birth Date",
        "Gender": "Gender",
        "Institution": "Institution",
        "Contact & address Detail": "Contact & address Detail",
        "Phone Number": "Phone Number",
        "Building": "Building",
        "Street": "Street",
        "City": "City",
        "State_place": "State",
        "Save Changes": "Save Changes",
        "Upload Image": "Upload Image",
        "Close": "Close",
        "Upload": "Upload",
        "Loading": "Loading",
        "Not a member ?": "Not a member ?",
        "Sign Up": "Sign Up",
        "Store": "Store",
        "Create Your": "Create Your",
        "and start exchanging with other students": "and start exchanging with other students",
        "Or Login Using:": "Or Login Using:",
        "Login": "Login",
        "LOGIN": "LOGIN ",
        "Already a member ?": "Already a member ?",
        "Email and password are required": "Email and password are required",
        "Password should be at least 8 characters long": "Password should be at least 8 characters long",
        "and Conditions and Privacy Policy of The website": "and Conditions and Privacy Policy of The website",
        "Email": "Email",
        "Password": "Password",
        "Full Name": "Full Name",
        "Address": "Address",
        "and Privacy Policy": "and Privacy Policy",
        "I have read and agree with all of the Terms": "I have read and agree with all of the Terms",
        "PROFILE": "PROFILE",
        "SECURITY": "SECURITY",
        "Upload ID": "Upload ID",
        "Edit ID": "Edit ID",
        "Edit Image": "Edit Image",
        "WHY DONATE": "WHY DONATE",
        "MAKE A DIFFERENCE BY DONATING": "MAKE A DIFFERENCE BY DONATING",
        "Donate paragraph": "In our fast-paced lives, it's easy to forget the power of gratitude. Students, often on tight budgets, rely on our support. Your donation is more than just financial aid; it's a symbol of gratitude. By giving generously, you're helping students create a brighter future, one contribution at a time.",
        "Small Help": "Small Help",
        "Some Help": "Some Help",
        "Big Help": "Big Help",
        "List an Item": "List an Item",
        "Drag and drop some files here or click to select files": "Drag and drop some files here or click to select files",
        "Location": "Location",
        "Category": "Category",
        "books": "books",
        "kitchen": "kitchen",
        "sports": "sports",
        "games": "games",
        "clothes": "clothes",
        "electronics": "electronics",
        "furniture": "furniture",
        "description": "Description",
        "Product Condition": "Product Condition",
        "Sell": "Sell",
        "Price Per Week": "Price Per Week",
        "Price": "Price",
        "Upload Image(s)": "Upload Image(s)",
        "List": "List",
        "Product Name": "Product Name",
        "Comments": "Comments",
        "Type your comment": "Type your comment",
        "Type your reply": "Type your reply",
        "Respond": "Respond",
        "Like": "Like",
        "Send": "Send",
        "Authentication error": "Invalid login credentials. Please check your email and password.",
        "Filter": "Filter",
        "Pending": "Pending",
        "Quick Links": "Quick Links",
        "About LACITÉ": "ABOUT LACITÉ",
        "About Us": "ABOUT US",
        "Github Project Link": "GITHUB PROJECT LINK",
        "Contact Us": "Contact Us",
        "We Couldn't Find the page you're looking for": "We Couldn't Find the page you're looking for",
        "Go To Home": "Go To Home",
        "We Couldn't Find the requested Product": "We Couldn't Find the requested Product."
    }
}
