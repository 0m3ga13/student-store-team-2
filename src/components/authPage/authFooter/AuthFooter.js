"use client";
import Link from "next/link";
import { useTranslations } from "next-intl";

const AuthFooter = ({ type }) => {
    const t = useTranslations("Index");
    const text = type === "LOGIN" ? "Not a member ?" : "Already a member ?";
    const link = type === "LOGIN" ? "/sign-up" : "/sign-in";
    const linkText = type === "LOGIN" ? "Sign Up" : "Login";

    return (
        <div>
            {" "}
            <div className='text-center m-2 text-sm   text-content'>
                {" "}
                {t(text)}
                <Link className='text-accent' href={link}>
                    {" "}
                    {t(linkText)}{" "}
                </Link>
            </div>
            {/*Buttons*/}
            <div className='flex justify-center items-center text-sm  m-2 text-content font-lato'>
                {t("Or:")}
            </div>
        </div>
    );
};

export default AuthFooter;
