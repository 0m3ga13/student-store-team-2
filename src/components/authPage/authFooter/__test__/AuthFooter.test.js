import { NextIntlClientProvider } from "next-intl";
import renderer from "react-test-renderer";

import AuthFooter from "../AuthFooter";
import messages from "../../../../../messages/en.json";

it("renders correctly", () => {
    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <AuthFooter />
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
