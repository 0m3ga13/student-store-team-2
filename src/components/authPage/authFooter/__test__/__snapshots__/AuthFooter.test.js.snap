// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
   
  <div
    className="text-center m-2 text-sm   text-content"
  >
     
    Already a member ?
    <a
      className="text-accent"
      href="/sign-in"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
       
      Login
       
    </a>
  </div>
  <div
    className="flex justify-center items-center text-sm  m-2 text-content font-lato"
  >
    Or:
  </div>
</div>
`;
